uefi/src/proto/device_path/mod.rs

```diff
@@ -176,11 +188,29 @@ impl DevicePathNode {
         self.full_type() == (DeviceType::END, DeviceSubType::END_ENTIRE)
     }
 
+    /// True if this node ends an entire [`DevicePath`].
+    #[must_use]
+    pub fn data(&self) -> &[u8] {
+        &self.data
+    }
+
     /// Convert from a generic [`DevicePathNode`] reference to an enum
     /// of more specific node types.
     pub fn as_enum(&self) -> Result<DevicePathNodeEnum, NodeConversionError> {
         DevicePathNodeEnum::try_from(self)
     }
+
+    /// If the device path node is of type (0x4, 0x4), it returns the underlying
+    /// data as [`CStr16`].
+    pub fn data_as_filename(&self) -> Option<Result<&CStr16, FromSliceWithNulError>> {
+        if self.device_type() == DeviceType::MEDIA
+            && self.sub_type() == DeviceSubType::HARDWARE_VENDOR
+        {
+            Some(CStr16::from_u8_with_nul(self.data()))
+        } else {
+            None
+        }
```

Comment:
Too bad that `.then` is not stable. :(

Reply:
`then` is stable, just not for const contexts

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7f625f39589b3e2ed13710d40adbb8f6